library/kani/src/models/mod.rs

```diff
@@ -0,0 +1,192 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+//! Contains definitions that Kani compiler may use to model functions that are not suitable for
+//! verification or functions without a body, such as intrinsics.
+//!
+//! Note that these are models that Kani uses by default; thus, we keep them separate from stubs.
+
+// Definitions in this module are not meant to be visible to the end user, only the compiler.
+#[allow(dead_code)]
+mod intrinsics {
+    use std::fmt::Debug;
+    use std::mem::size_of;
+
+    /// Similar definition to portable SIMD.
+    /// We cannot reuse theirs since TRUE and FALSE defs are private.
+    /// We leave this private today, since this is not necessarily a final solution, so we don't
+    /// want users relying on this.
+    /// Our definitions are also a bit more permissive to comply with the platform intrinsics.
+    pub(super) trait MaskElement: PartialEq + Debug {
+        const TRUE: Self;
+        const FALSE: Self;
+    }
+
+    macro_rules! impl_element {
+        { $ty:ty } => {
+            impl MaskElement for $ty {
+                const TRUE: Self = -1;
+                const FALSE: Self = 0;
+            }
+        }
+    }
+
+    macro_rules! impl_unsigned_element {
+        { $ty:ty } => {
+            impl MaskElement for $ty {
+                const TRUE: Self = <$ty>::MAX;
```

Comment:
Any reference for the value used here?

Reply:
I was actually debating if we should accept unsigned input, but the rust compiler and MIRI seems to be OK with that. Anyway, there isn't much when it comes to references to intrinsics and the portable_simd library restricts the usage of Masks to just a few options, and none is unsigned.